linux/datadog_wrapper

```diff
@@ -1,96 +1,165 @@
+setUpBinary() {
+    BINARY="$1"
+    ARCHIVED_BINARY="$BINARY-${DD_AAS_LINUX_VERSION}.tar.gz"
 
-    ORIG_NODE_OPTIONS=$NODE_OPTIONS
-    export NODE_OPTIONS="--require=$DD_DIR/node_modules/dd-trace/init $ORIG_NODE_OPTIONS"
+    check_file_exists_or_curl "$ARCHIVED_BINARY" "$BINARY"
+    # do not start the binary if curling failed
+    if [ "$?" -ne 0 ]; then
+        echo "Curling $BINARY was unsuccesful and the binary will not be initialized"
+        return
+    fi
 
-    # confirm updates to NODE_OPTIONS
-    node --help >/dev/null || (export NODE_OPTIONS=$ORIG_NODE_OPTIONS && return)
+    echo "Starting $BINARY"
+    if [ $BINARY = "dogstatsd" ]; then
+        ./$BINARY start &
+        trap 'kill $! : exit' INT
+        return
+    fi
+
+    ./"$1" &
+    trap 'kill $! : exit' INT
 }
 
-setUpDotnetEnv() {
-    echo "Setting up Datadog tracing for .NET"
+check_file_exists_or_curl() {
+    # We don't want to curl the binary if the same version is already present in AAS
+    # $1 is the remote, archived and versioned binary on github, eg trace-agent-${DD_AAS_LINUX_VERSION}.tar.gz
+    # $2 is the local binary after it is unarchived
+    ARCHIVED_BINARY="$1"
+    UNARCHIVED_BINARY="$2"
 
-    setUpCommonEnv || return
+    if [ ! -f "$ARCHIVED_BINARY" ]; then
 
-    if [ -z "$DD_DOTNET_TRACER_VERSION" ]; then
-        DD_DOTNET_TRACER_VERSION=2.19.0
+        if [ -f "$UNARCHIVED_BINARY" ]; then
+            echo "Removing old versions of the $UNARCHIVED_BINARY binary"
+            rm "$UNARCHIVED_BINARY*"
+        fi
+
+        echo "Downloading required Datadog binaries from $DD_BINARIES_URL/$DD_AAS_LINUX_VERSION/$ARCHIVED_BINARY"
+        curl_or_log_failure "$DD_BINARIES_URL/$DD_AAS_LINUX_VERSION/$ARCHIVED_BINARY" "$ARCHIVED_BINARY"
+    else
+        echo "Version $DD_AAS_LINUX_VERSION previously installed for the $UNARCHIVED_BINARY binary"
     fi
-    DD_DOTNET_TRACER_FILE=datadog-dotnet-apm-${DD_DOTNET_TRACER_VERSION}.tar.gz
-    DD_DOTNET_TRACER_URL=https://github.com/DataDog/dd-trace-dotnet/releases/download/v${DD_DOTNET_TRACER_VERSION}/${DD_DOTNET_TRACER_FILE}
+}
 
-    echo "Installing .NET tracer from $DD_DOTNET_TRACER_URL"
-    curl -L $DD_DOTNET_TRACER_URL -o $DD_DOTNET_TRACER_FILE|| return
-    tar -xzf $DD_DOTNET_TRACER_FILE || return
+curl_or_log_failure() {
+    # $1 is the URL of the archived binary
+    # $2 is the unarchived filename we expect, eg. trace-agent
+    ARCHIVED_BINARY_URL="$1"
+    ARCHIVED_BINARY="$2"
 
-    export CORECLR_ENABLE_PROFILING=1
-    export CORECLR_PROFILER={846F5F1C-F9AE-4B07-969E-05C26BC060D8}
-    export CORECLR_PROFILER_PATH=${DD_DIR}/Datadog.Trace.ClrProfiler.Native.so
-    export DD_DOTNET_TRACER_HOME=${DD_DIR}
+    if curl -L --fail "$ARCHIVED_BINARY_URL" -o "$ARCHIVED_BINARY"; then
+        echo "Decompressing $ARCHIVED_BINARY_URL"
+        tar -zxf "$ARCHIVED_BINARY" || return
+    else
+        echo "Failed to download $ARCHIVED_BINARY succesfully from $ARCHIVED_BINARY_URL."
```

Comment:
If the binary fails to download and then this line of code executes, the `curl_or_log_failure` function will have a return code of `0`. Therefore, we'd still attempt to install the tracer, even if this fails.